server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -225,11 +275,24 @@ public void onTimeout(TimeValue timeout) {
         }
     }
 
+    private static final long DELAY_MILLIS = 100L;
+
+    // At the most we wait for 5s for a master node request to be retried if master returned a circuit breaker exception
+    private static final long MAX_DELAY = TimeValue.timeValueSeconds(5).millis();
+
     /**
-     * Allows to conditionally return a different master node action name in the case an action gets renamed.
-     * This mainly for backwards compatibility should be used rarely
+     * Computes the next delay from the number retries that have already happened due to consecutive {@link CircuitBreakingException} on
+     * a remote master node.
+     *
+     * @param currentRetry number of retries including the current one that have already happened
+     * @return retry delay in ms
      */
-    protected String getMasterActionName(DiscoveryNode node) {
```

Comment:
This wasn't overriden anywhere any longer so I just removed to to keep the new retry logging simpler